lmfdb/knowledge/main.py

```diff
@@ -41,7 +41,7 @@
 
 # know IDs are restricted by this regex
 allowed_knowl_id = re.compile("^[a-z0-9._-]+$")
-allowed_annotation_id = re.compile("^[a-zA-Z0-9._-~]+$")  # all unreserved URL characters
+allowed_annotation_id = re.compile(r"^[a-zA-Z0-9._\-~]+$")  # all unreserved URL characters
```

Comment:
While `\` is not reserved, it is weirdly treated.

Reply:
The problem is that `-` has a special meaning inside brackets, so this previously was including all the characters between `_` and `~`.  Instead, we want the three characters `_`, `-` and `~`, which is achieved by escaping `-`.